Kill queries

zhenlineo opened the pull request:
Added `session#reset` to stop long running statements and reset the session to a clean state

Commit:
Added `session#kill` to stop running more statements and reset the session to a clean state

technige commented on driver/src/main/java/org/neo4j/driver/v1/Session.java:
```diff
@@ -58,7 +58,10 @@
+    /**
+     * Stop running more statements in this session and rest the session to a clean state.
+     */
+    void kill();
```
As per earlier conversation, this should be named `reset` and could be described as follows:

```
/**
 * Reset the current session. This sends an immediate RESET signal to the server which both interrupts
 * any statement that is currently executing and ignores any subsequently queued statements. Following
 * the reset, the current transaction will have been rolled back and any outstanding failures will
 * have been acknowledged.
 */
void reset();
```

Commit:
Renamed the kill method to reset
Added test using long running procedures to test session.reset
Made the client to not ack_failure while a reset is called asynclly but has not yet received success to avoid ack_failure on IDEL state in server
